### PR TITLE
feat: workflows pilot

### PR DESCRIPTION
**Pull request by jhackett1**
satisfies https://trello.com/c/1HyoOcl9/124-make-a-pull-request-to-the-mainstream-tool.

makes the workflows [the workflows app](https://github.com/lBHackney-IT/lbh-core-pathway-pilot) accessible to users of the mainstream tool, for the sake of the pilot.

there is no deep integration - just simple html links back and forth.

it adds:

- [x] "workflows" link on the dashboard, for all users
- [x] "workflows" link on the person view to all that person's workflows
- [x] "start new assessment, support plan or workflow" link to the "add something new" dialog
- [x] make that dialog link visible for users in the pilot google group _only_
- [x] a dismissable, one-off "welcome to the pilot" onboarding dialog, for users in the pilot google group only

all is behind the `workflows-pilot` feature flag.

### CODE REVIEW

File: components/AddFormDialog/AddFormDialog.spec.tsx [hunk 1]
```diff
@@ -24,11 +25,13 @@ describe('AddFormDialog', () => {
   it('shows forms from two sources', () => {
     render(
       <AuthProvider user={mockedUser}>
-        <AddFormDialog
-          isOpen={true}
-          onDismiss={jest.fn()}
-          person={mockedResident}
-        />
+        <FeatureFlagProvider features={{}}>
```
Comment (chubberlisk) [inline components/AddFormDialog/AddFormDialog.spec.tsx:28]: Add additional tests around the feature flag and eventually the Google group.

File: components/AddFormDialog/AddFormDialog.tsx [hunk 2]
```diff
@@ -65,21 +68,35 @@ const AddFormDialog = ({
...
+              system: false,
+              groupRecordable: false,
+              approvable: false,
+              inPreview: false,
+            }))
+            .concat(
+              isFeatureActive('workflows-pilot')
+                ? {
+                    label: 'Assessment, support plan or workflow',
+                    href: `${process.env.NEXT_PUBLIC_WORKFLOWS_PILOT_URL}/workflows/new?social_care_id=${person.id}`,
```
Comment (chubberlisk) [inline components/AddFormDialog/AddFormDialog.tsx:90]: We'll need to add this to the serverless config and a new SSM parameter in AWS.

File: components/AddFormDialog/AddFormDialog.tsx [hunk 3]
```diff
@@ -65,21 +68,35 @@ const AddFormDialog = ({
...
+                user.email,
+                f.value
+              )}`,
+              system: false,
+              groupRecordable: false,
+              approvable: false,
+              inPreview: false,
+            }))
+            .concat(
+              isFeatureActive('workflows-pilot')
```
Comment (chubberlisk) [inline components/AddFormDialog/AddFormDialog.tsx:87]: Update this so only users in the Google group can see this.

File: components/Dashboard/DashboardWrapper.spec.tsx [hunk 4]
```diff
@@ -11,9 +12,11 @@ jest.mock('next/router', () => ({
 describe(`DashboardWrapper`, () => {
   it('should render properly', () => {
     const { asFragment } = render(
-      <DashboardWrapper>
-        <div>foo</div>
-      </DashboardWrapper>
+      <FeatureFlagProvider features={{}}>
```
Comment (chubberlisk) [inline components/Dashboard/DashboardWrapper.spec.tsx:15]: Add additional tests around feature flag.

File: components/NewPersonView/Layout.tsx [hunk 5]
```diff
@@ -190,6 +190,16 @@ const Layout = ({ person, children }: Props): React.ReactElement => {
                   {link.text}
                 </NavLink>
               ))}
+
+              <ConditionalFeature name="workflows-pilot">
```
Comment (chubberlisk) [inline components/NewPersonView/Layout.tsx:194]: Add tests around this.

File: components/OnboardingDialog/index.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react';
+import Dialog from 'components/Dialog/Dialog';
+
+const OnboardingDialog = (): React.ReactElement | null => {
```
Comment (chubberlisk) [inline components/OnboardingDialog/index.tsx:4]: Add tests for this component.

File: components/AddFormDialog/AddFormDialog.spec.tsx (hunk 1 above)
Comment (chubberlisk) [reply components/AddFormDialog/AddFormDialog.spec.tsx]: Added tests and will only display if feature flag is on and user is in workflow pilot.

File: components/AddFormDialog/AddFormDialog.tsx (hunk 3 above)
Comment (chubberlisk) [reply components/AddFormDialog/AddFormDialog.tsx]: Updated so will only display if feature flag is on and user is in workflow pilot.

File: components/Dashboard/DashboardWrapper.spec.tsx (hunk 4 above)
Comment (chubberlisk) [reply components/Dashboard/DashboardWrapper.spec.tsx]: Added tests.

File: components/NewPersonView/Layout.tsx (hunk 5 above)
Comment (chubberlisk) [reply components/NewPersonView/Layout.tsx:194]: Added tests and also only displays if user is in the workflows pilot.

File: components/OnboardingDialog/index.tsx (hunk 6 above)
Comment (chubberlisk) [reply components/OnboardingDialog/index.tsx:4]: Added tests.

File: components/AddFormDialog/AddFormDialog.tsx [hunk 7]
```diff
@@ -65,22 +67,36 @@ const AddFormDialog = ({
...
+                user.email,
+                f.value
+              )}`,
+              system: false,
+              groupRecordable: false,
+              approvable: false,
+              inPreview: false,
+            }))
+            .concat(
+              isFeatureActive('workflows-pilot') && user.isInWorkflowsPilot
```
Comment (jhackett1) [inline components/AddFormDialog/AddFormDialog.tsx:86]: a suggestion - is it possible to check the current user in the feature flag config?

if so, we could simply use `isFeatureActive` here to keep things clean, and keep the two-part logic of whether the feature is active inside the flag where it belongs:
1. what environment are we on?
2. what group is this user in?

Comment (chubberlisk) [reply components/AddFormDialog/AddFormDialog.tsx:86]: Potentially yeah, but my understanding is that we want to show the Workflows link in the dashboard for everyone but the add something new link we only want to show for users in the pilot so I think it needs to link outside of it.

**Comment by acodeninja** [comment]
So everything here looks fine to me, I approve, but I would prefer someone who works on this tool day-to-day to get the final say on if it is mergeable.

**Review by mapleleafjack — APPROVED** [review]
Looks good to me!

**Pull request merged**